src/types/redis_stream.cc

```diff
@@ -343,8 +346,26 @@ rocksdb::Status Stream::DeletePelEntries(const Slice &stream_name, const std::st
     std::string entry_key = internalPelKeyFromGroupAndEntryId(ns_key, metadata, group_name, id);
     std::string value;
     s = storage_->Get(rocksdb::ReadOptions(), stream_cf_handle_, entry_key, &value);
+
     if (s.ok()) {
       *acknowledged += 1;
+      StreamPelEntry pel_entry = decodeStreamPelEntryValue(value);
+
+      const std::string &consumer_name = pel_entry.consumer_name;
+      std::string consumer_key = internalKeyFromConsumerName(ns_key, metadata, group_name, consumer_name);
+      std::string get_consumer_value;
+      s = storage_->Get(rocksdb::ReadOptions(), stream_cf_handle_, consumer_key, &get_consumer_value);
+      if (!s.ok() && !s.IsNotFound()) {
+        return s;
+      }
+      if (s.IsNotFound()) {
+        return rocksdb::Status::OK();
```

Comment:
I don't think it's a return here. The above line is the same. It should not affect other consumers and the final result.